docs/csharp/language-reference/statements/lock.md

```diff
@@ -43,17 +54,17 @@ You can't use the [`await` expression](../operators/await.md) in the body of a `
 
 ## Guidelines
 
-When you synchronize thread access to a shared resource, lock on a dedicated object instance (for example, `private readonly object balanceLock = new object();`) or another instance that is unlikely to be used as a lock object by unrelated parts of the code. Avoid using the same lock object instance for different shared resources, as it might result in deadlock or lock contention. In particular, avoid using the following instances as lock objects:
+Beginning with .NET 9 and C# 13, lock a dedicated object instance of the <xref:System.Threading.Lock?displayProperty=nameWithType> type. That ensures the best performance. In addition, the compiler issues a warning if a known `Lock` object is cast to another type and locked. If using an older version of .NET and C#, lock on a dedicated object instance that isn't used for another purpose. Avoid using the same lock object instance for different shared resources, as it might result in deadlock or lock contention. In particular, avoid using the following instances as lock objects:
```

Comment:
```suggestion
Beginning with .NET 9 and C# 13, lock a dedicated object instance of the <xref:System.Threading.Lock?displayProperty=nameWithType> type for best performance. In addition, the compiler issues a warning if a known `Lock` object is cast to another type and locked. If using an older version of .NET and C#, lock on a dedicated object instance that isn't used for another purpose. Avoid using the same lock object instance for different shared resources, as it might result in deadlock or lock contention. In particular, avoid using the following instances as lock objects:
```